### ISSUE TITLE
Mail action sends email for jail stops but not starts or bans

### ISSUE THREAD

**Issue by dsmithdewarcom** [fail2ban/fail2ban#2146]
- Fail2Ban version (including any possible distribution suffixes):0.10.1-1
- OS, including release name/version: Ubuntu 16.04.4

### The issue:
Following upgrade from 0.08.11-1, the action "%(action_mw)s" only sending emails for jail stops [reloads], not for jail starts and bans

#### Steps to reproduce
Apply "action = %(action_mw)s" to selct jail(s) -- in my case, "recidive" and a custom jail, "apache-adminrsc"

#### Expected behavior
email notification of jail ban|start|stop.

#### Observed behavior
email notification of jail stop only.

#### Any customizations done to /etc/fail2ban/ configuration
(as reported in earlier issues)

#### Relevant parts of /var/log/fail2ban.log file:
(email notifications aren't logged)

**Comment by sebres** [comment]
The parameter `%(action_mw)s` is an interpolation variable (so depended on your config/distribution)...
Which mail-action it's exactly on your system?
```
fail2ban-client -d | grep <some-jail-with-action_mw>
```
Take a look on action.

I don't have fully understood what do you mean by "but not starts or bans", but possibly not a bug (but feature), if the handling is affected by RFE #1669. 
But this should avoid the notification for already banned tickets after restart/reload (which called restored there)...

**Comment by dsmithdewarcom** [comment]
Perhaps there is another update that is causing the issue that I need to reconfigure for, as it wasn't present in 0.8.11-1, however I did not edit any actions in either version.

(actual email address used in "jail.local" changed here to "email@mydomain.com")
```
$ fail2ban-client -d | grep recidive
 No file(s) found for glob /var/log/nginx/*access.log
['add', 'recidive', 'auto']
['set', 'recidive', 'addfailregex', '^(?:\\[\\])?\\s*(?:<[^.]+\\.[^.]+>\\s+)?(?:\\S+\\s+)?(?:kernel: \\[ *\\d+\\.\\d+\\]\\s+)?(?:@vserver_\\S+\\s+)?(?:(?:(?:\\[\\d+\\])?:\\s+[\\[\\(]?(?:fail2ban(?:-server|\\.actions)\\s*)(?:\\(\\S+\\))?[\\]\\)]?:?|[\\[\\(]?(?:fail2ban(?:-server|\\.actions)\\s*)(?:\\(\\S+\\))?[\\]\\)]?:?(?:\\[\\d+\\])?:?)\\s+)?(?:\\[ID \\d+ \\S+\\]\\s+)?(?:\\s*fail2ban\\.actions\\s*(?:\\[\\d+\\])?:\\s+)?NOTICE\\s+\\[(?!recidive\\])(?:.*)\\]\\s+Ban\\s+<HOST>\\s*$']
['set', 'recidive', 'addjournalmatch', '_SYSTEMD_UNIT=fail2ban.service', 'PRIORITY=5']
['set', 'recidive', 'datepattern', '^{DATE}']
['set', 'recidive', 'usedns', 'warn']
['set', 'recidive', 'addlogpath', '/var/log/fail2ban.log', 'head']
['set', 'recidive', 'maxretry', 3]
['set', 'recidive', 'addignoreip', '192.168.0.1/24']
['set', 'recidive', 'logencoding', 'auto']
['set', 'recidive', 'bantime', '1w']
['set', 'recidive', 'ignorecommand', '']
['set', 'recidive', 'findtime', '3d']
['set', 'recidive', 'addaction', 'iptables-allports']
['multi-set', 'recidive', 'action', 'iptables-allports', [['actionunban', '<iptables> -D f2b-recidive -s <ip> -j <blocktype>'], ['actionflush', '<iptables> -F f2b-recidive'], ['actionstop', '<iptables> -D INPUT -p tcp -j f2b-recidive\n<iptables> -F f2b-recidive\n<iptables> -X f2b-recidive'], ['actionstart', '<iptables> -N f2b-recidive\n<iptables> -A f2b-recidive -j RETURN\n<iptables> -I INPUT -p tcp -j f2b-recidive'], ['actionban', '<iptables> -I f2b-recidive 1 -s <ip> -j <blocktype>'], ['actioncheck', "<iptables> -n -L INPUT | grep -q 'f2b-recidive[ \\t]'"], ['protocol', 'tcp'], ['chain', 'INPUT'], ['lockingopt', '-w'], ['blocktype', 'REJECT --reject-with icmp-port-unreachable'], ['blocktype?family=inet6', 'REJECT --reject-with icmp6-port-unreachable'], ['port', 'ssh,http,https'], ['actname', 'iptables-allports'], ['bantime', '1w'], ['iptables', 'iptables <lockingopt>'], ['returntype', 'RETURN'], ['iptables?family=inet6', 'ip6tables <lockingopt>'], ['name', 'recidive']]]
['set', 'recidive', 'addaction', 'mail-whois']
['multi-set', 'recidive', 'action', 'mail-whois', [['actionstop', 'printf %b "Hi,\\n\nThe jail recidive has been stopped.\\n\nRegards,\\n\nFail2Ban"|mail -s "[Fail2Ban] recidive: stopped on <fq-hostname>" email@mydomain.com'], ['actioncheck', ''], ['norestored', True], ['actionban', 'printf %b "Hi,\\n\nThe IP <ip> has just been banned by Fail2Ban after\n<failures> attempts against recidive.\\n\\n\nHere is more information about <ip> :\\n\n`whois <ip> || echo "missing whois program"`\\n\nRegards,\\n\nFail2Ban"|mail -s "[Fail2Ban] recidive: banned <ip> from <fq-hostname>" email@mydomain.com'], ['actionstart', 'printf %b "Hi,\\n\nThe jail recidive has been started successfully.\\n\nRegards,\\n\nFail2Ban"|mail -s "[Fail2Ban] recidive: started on <fq-hostname>" email@mydomain.com'], ['actionunban', ''], ['protocol', 'tcp'], ['name', 'recidive'], ['chain', 'INPUT'], ['dest', 'email@mydomain.com'], ['actname', 'mail-whois'], ['sender', 'Fail2Ban']]]
['start', 'recidive']
```

**Comment by sebres** [comment]
So this is `mail-whois` which is used for you...
Corresponding your config, it looks like:
- start:
```bash
printf %b "Hi,\n
The jail recidive has been started successfully.\n
Regards,\n
Fail2Ban"|mail -s "[Fail2Ban] recidive: started on <fq-hostname>" email@mydomain.com
```
- ban:
```bash
printf %b "Hi,\n
The IP <ip> has just been banned by Fail2Ban after
<failures> attempts against recidive.\n\n
Here is more information about <ip> :\n
`whois <ip> || echo "missing whois program"`\n
Regards,\n
Fail2Ban"|mail -s "[Fail2Ban] recidive: banned <ip> from <fq-hostname>" email@mydomain.com
```
- stop:
```bash
printf %b "Hi,\n
The jail recidive has been stopped.\n
Regards,\n
Fail2Ban"|mail -s "[Fail2Ban] recidive: stopped on <fq-hostname>" email@mydomain.com
```

I don't see something wrong why start/ban should not work (tags `<ip>` and `<fq-hostname>` will be replaced before execution with IP of the ticket and FQDN-hostname).

Do you see any error in fail2ban.log (by start/ban inside recidive jail)?

**Comment by dsmithdewarcom** [comment]
No errors in log -- fail2ban.log last reload ("[...]" = unrelated lines):
```
2018-05-30 10:31:40,159 fail2ban.server         [18373]: INFO    Reload all jails
[...]
2018-05-30 10:31:40,352 fail2ban.server         [18373]: INFO    Jail recidive is not a JournalFilter instance
2018-05-30 10:31:40,379 fail2ban.filter         [18373]: INFO      maxRetry: 3
2018-05-30 10:31:40,380 fail2ban.filter         [18373]: INFO      encoding: UTF-8
2018-05-30 10:31:40,382 fail2ban.actions        [18373]: INFO      banTime: 604800
2018-05-30 10:31:40,384 fail2ban.filter         [18373]: INFO      findtime: 259200
[...]
2018-05-30 10:31:40,891 fail2ban.server         [18373]: INFO    Jail 'recidive' reloaded
[...]
2018-05-30 10:31:41,457 fail2ban.server         [18373]: INFO    Reload finished.
```
mail stuff in jail.local:
# ACTIONS

```
mta = mail
destemail = email@mydomain.com
sender = Fail2Ban
```
recidive jail (jail.local):
```
[recidive]
enabled  = true
logpath  = /var/log/fail2ban.log
action   = %(action_mw)s
protocol = tcp
port     = ssh,http,https
bantime  = 1w
findtime = 3d
```
(filter.d/recidive.conf unedited from original installation)

**Comment by dsmithdewarcom** [comment]
I understand about the reloads/already notified stuff, which is great.

The good news is I'm getting a lot fewer "recidive" bans than in the past, so I think you can take some credit for that :)

I did get a new recidive ban earlier this morning but haven't received an email notification for:

2018-05-31 02:01:06,161 fail2ban.actions        [18373]: NOTICE  [recidive] Ban 219.155.47.54

My main concern though is with the lack of jail "start" notifications, which I now have no way of confirming without remote logging in to the server following a jail "stop" email.

I do have a high level of confidence that the jail *probably* restarted anyway, but the point of email notification is to eliminate all doubt, yes?

**Comment by sebres** [comment]
> but haven't received an email notification for `2018-05-31 02:01:06 ... [recidive] Ban 219.155.47.54`

So do you see this time any error message after ban-message in fail2ban.log (round about 2018-05-31 02:01:06)?
If not, could you check please also the log of your mail-subsystem (mail-command)? 
Just to be sure the mail was sent (but possibly not received from you)... Something like filter/spam-folder/etc.

**Comment by dsmithdewarcom** [comment]
No errors in fail2ban.log and mail.err (which is empty), mail.log shows only one mail sent today from server for a cron job (which I received).

**Comment by dsmithdewarcom** [comment]
Whelp, I got a jail start email this morning. After a reload fail (due to un-escaped backslash added to regex) I executed a "fail2ban-client reload" command to restart the jail

![screenshot from 2018-06-01 09-37-55](https://user-images.githubusercontent.com/9698006/40843784-18794f72-6580-11e8-97f0-ad2b0921d848.jpg)

So that's apparently working after all. What's confusing is that a jail reload begets a stop email, but not a start email. 

Also never sent a ban email for the one recidive ban from yesterday as reported above.
